File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& arr, int k) {
        int n = arr.size();
    if (n == 0)
    return;
  k = k % n;
  if(k == 0) return;
  int temp[k];
  for (int i = n-k; i < n; i++)
  {
    temp[i-n+k] = arr[i];
  }
  for (int i = n-1 ; i >= k ; i--)
  {
    arr[i] = arr[i - k];
  }
  for (int i = 0; i < k; i++)
  {
    arr[i] = temp[i];
  }
    }
};

File: 0189-rotate-array/0189-rotate-array.rs
impl Solution {
  pub fn rotate(nums: &mut Vec<i32>, k: i32) {
        let len = nums.len();
        if len == 0 {
            return;
        }
        
        let k = (k as usize) % len; // Handle cases where k is larger than the array size
        let mut nums2 = vec![0; len];
        
        for i in 0..k {
            nums2[i] = nums[len - k + i];
        }
        
        for i in 0..(len - k) {
            nums2[k + i] = nums[i];
        }
        
        for i in 0..len {
            nums[i] = nums2[i];
        }
    }
}